File: AllMyBooksArePacked/Entities/Box.rb
class Box
  attr_accessor :id, :totalWeight, :contents
  def initialize (id=nil, totalWeight=0.0, contents=[])
    @id = id
    @totalWeight = totalWeight
    @contents = contents
  end
  def update_box_weight
    @totalWeight = 0
    @contents.each do |book|
      @totalWeight += book.shipping_weight
    end
  end
  def to_hash
    contents = @contents.map {|book| book.to_hash}
    return {
      id: @id,
      totalWeight: @totalWeight,
      contents: contents
    }
  end
end

File: AllMyBooksArePacked/js/box.js
function Box(){
  this.totalWeight;
  this.contents = [];
}

Box.prototype.updateTotalWeight = function(){
  var combinedWeight = 0;
  for (var i = 0; i < this.contents.length; i++){
    combinedWeight += this.contents[i].shippingWeight;
  }
  this.totalWeight = combinedWeight.toFixed(2);
}